docs/topics/signals.rst

```diff
@@ -68,17 +68,15 @@ Let's take an example::
             crawler.signals.connect(spider.item_scraped, signal=signals.item_scraped)
             return spider
 
-        def item_scraped(self, item):
+        async def item_scraped(self, item):
             # Send the scraped item to the server
-            d = treq.post(
+            response = await treq.post(
                 'http://example.com/post',
                 json.dumps(item).encode('ascii'),
                 headers={b'Content-Type': [b'application/json']}
             )
-
-            # The next item will be scraped only after
-            # deferred (d) is fired
-            return d
+            content = await treq.content(response)
+            return content
```

Comment:
I don’t think these last 2 lines are correct. I believe, to reproduce the original behavior, you just need to return the result of `treq.post`, and let the caller of `item_scraped` await it. The original code does not wait for the response content, and does not return it.

@wRAR Thoughts?

Reply:
The old example: make a POST request, return a Deferred that fires when the request is completed. The result of that Deferred will be a response object.

The new example that does the same would be `return response`.

On the other hand, I don't think results of signal handlers, at least the `item_scraped` one, are used anywhere, so ending the method on just `await treq.post()` would do the same.

Reply:
@wRAR I fixed it as you suggested. Leaving the `return` statement just in case.